site/en/docs/extensions/reference/declarativeNetRequest/index.md

```diff
@@ -158,102 +157,144 @@ examples of URL filters:
+
+### Web accessible resources {: #implementation-web-accessible-resources }
+
+A declarativeNetRequest rule cannot redirect from a public resource request to a resource that is not web accessible. Doing so triggers an error. This is true even if the specified web accessible resource is owned by the redirecting extension. To declare resources for declarativeNetRequests, use the manifest's [`"web_accessible_resources"`](/docs/extensions/mv3/manifest/web_accessible_resources/) array.
+
+## Examples {: #example }
+
+### Code examples {: #code-examples }
+
+#### Update dynamic rules {: #update-dynamic-rule-examples }
+
+The following example shows how to call `updateDynamicRules()`. The procedure for `updateSessionRules()` is the same.
+
+```js
+// Get arrays containing new and old rules
+const newRules = getNewRules();
+const oldRules = await chrome.declarativeNetRequest.getDynamicRules();
+const oldRuleIds = oldRules.map(rule => rule.id);
+
+// Use the arrays to update the dynamic rules
+await chrome.declarativeNetRequest.updateDynamicRules({
+  removeRuleIds: oldRuleIds,
+  addRules: newRules
+});
+```
+
+#### Update static rulesets {: #update-static-rulesets }
+
+The following example shows how to enable and disable rulesets, while taking into account the number of available rulesets and the maximum number of enabled static rulesets. You would do this is in situations where the number of static rules you need exceeds the number allowed. For this to work, some of your rulesets should be installed with some of your rulesets disabled (setting `"Enabled"` to `false` with in the manifest file).
+
+```js
+function updateStaticRules(enableRulesetIds, disableCandidateIds) {
+  // Create the options structure for the call to updateEnabledRulesets()
+  let options = { enableRulesetIds: enableRulesetIds }
+  // Get the number of enabled static rules
+  const enabledStaticCount = await chrome.declarativeNetRequest.getEnabledRulesets();
+  // Compare rule counts to determine if anything needs to be disabled so that
+  // new rules can be enabled
+  const propsedCount = enableRulesetIds.length;
+  if (enabledStaticCount + proposedCount > chrome.declarativeNetRequest.MAX_NUMBER_OF_ENABLED_STATIC_RULESETS) {
+    options.disableRulesetIds = disableCandidateIds
+  }
+  // Update the enabled static rules
+  await chrome.declarativeNetRequest.updateEnabledRulesets(options);
 }
 ```
 
-{% Label %}rules.json:{% endLabel %}
+### Rule examples {: #rule-examples }
+The examples below illustrate how Chrome prioritizes rules in an extension. You may want to open the [prioritization](#implementation-matching-algorithm) rules in a separate window when reviewing them.
+
+#### The "priority" key {: #priority-key }
+
+These examples require the following host permission:
+
+`*://*.example.com/*`
```

Comment:
> `*://*.example.com/*`

Is this the only permission required for the example below? It might be nice to have a full manifest code sample for ease of copy/paste